include/RevOpts.h

```diff
@@ -18,15 +18,46 @@
 #include <cinttypes>
 #include <map>
 #include <string>
+#include <tuple>
+#include <type_traits>
 #include <utility>
 #include <vector>
 
 namespace SST::RevCPU {
 
 class RevOpts {
+
+  // Queries whether type is a std::vector type
+  template<typename T>
+  struct is_vector : std::false_type {};
+
+  template<typename T>
+  struct is_vector<std::vector<T>> : std::true_type {};
+
+  // Return a property by looking up its table
+  // VAL is a universal reference so that either an lvalue reference or an assignable rvalue reference such as std::tie can be used
+  template<typename MAP, typename VAL>
+  bool GetProperty( uint32_t Core, const MAP& Map, VAL&& Val ) const {
+    if constexpr( is_vector<MAP>::value ) {
+      // If MAP is a vector
+      return Core < numCores ? Val = Map[Core], true : false;
+    } else {
+      // If MAP is a map/unordered_map
+      auto it                      = Map.find( Core );
+      return it != Map.end() ? Val = it->second, true : false;
+    }
+  }
```

Comment:
Only one of the `if constexpr()` branches will be taken, and the other will be discarded at compile-time, and will produce no errors.

So we can write one function which operates differently depending on whether `MAP` is a vector or not.

`VAL&& Val` is a _universal reference_, meaning `VAL` will be of lvalue reference type for lvalue arguments (arguments with a name whose address can be taken) and `VAL` will be of rvalue reference type for rvalue arguments (arguments with no name or address, which represent temporary expressions which are fleeting and whose resources can be moved instead of deep-copied). Inside `GetProperty`, `Val` is an lvalue even if it bound to an rvalue. `std::tie()` creates an rvalue tuple of references, and can be assigned to even though it is an rvalue, in which case it assigns to its lvalue reference elements.